packages/ag-charts-enterprise/src/features/annotations/states/textualStartEndState.ts

```diff
@@ -0,0 +1,132 @@
+        const onStartClick = ({ point }: { point: Point }) => {
+            const datum = this.createDatum();
+            datum.set({ start: point, end: point, text: '' });
+            ctx.create(datum);
+            ctx.resetToolbarButtonStates();
+        };
+
+        const onEndHover = ({ point }: { point: Point }) => {
+            ctx.datum()?.set({ end: point });
+            ctx.node()?.toggleActive(true);
+            ctx.node()?.toggleHandles({ end: false });
+            ctx.update();
+        };
+
+        const onEndClick = ({ point }: { point: Point }) => {
+            ctx.datum()?.set({ end: point });
+            ctx.node()?.toggleHandles({ end: true });
+        };
+
+        const onSave = ({ textInputValue }: { textInputValue?: string }) => {
+            if (textInputValue !== null) {
+                ctx.datum()?.set({ text: textInputValue });
+                ctx.update();
+            } else {
+                ctx.delete();
+            }
+        };
+
+        super('start', {
+            start: {
+                click: {
+                    target: 'waiting-first-render',
+                    action: onStartClick,
+                },
+                cancel: StateMachine.parent,
+            },
+            end: {
+                hover: onEndHover,
+                click: {
+                    target: 'edit',
+                    action: onEndClick,
+                },
+                cancel: StateMachine.parent,
+            },
+            'waiting-first-render': {
+                render: {
+                    target: 'end',
+                    action: () => {
+                        ctx.node()?.toggleActive(true);
+                        ctx.showAnnotationOptions();
+                    },
+                },
+            },
+            edit: {
+                onEnter: () => {
+                    ctx.showTextInput();
+                    const datum = ctx.datum();
+                    if (datum) {
+                        datum.visible = false;
+                    }
+                },
+                textInput: ({ bbox }) => {
+                    const node = ctx.node();
+                    node?.setTextInputBBox(bbox);
+                    ctx.update();
+                },
+                keyDown: [
+                    {
+                        guard: ({ key }: { key: string }) => key === 'Escape',
+                        target: StateMachine.parent,
+                        action: () => {
+                            ctx.delete();
+                        },
+                    },
+                    {
+                        guard: ({ key, shiftKey }: { key: string; shiftKey: boolean }) => !shiftKey && key === 'Enter',
+                        target: StateMachine.parent,
+                        action: onSave,
+                    },
+                    {
+                        target: 'edit',
+                        action: onSave,
+                    },
```

Comment:
We don't want this one anymore as in `TextualPointState`, as we don't need to read the size from the label bbox. And we shouldn't save until they press enter, so the escape key works properly.